File: Go/Go_100-999/485.go
// Go
// leetcode 485
// https://leetcode.cn/problems/max-consecutive-ones/

func findMaxConsecutiveOnes(nums []int) int {
	cur, res := 0, 0
	for _, v := range nums {
		if v == 1 {
			cur++
			if cur > res {
				res = cur
			}
		} else {
			cur = 0
		}
	}
	return res
}

File: cpp/cpp_100-999/485.cc
// C++
// leetcode 485
// https://leetcode.cn/problems/max-consecutive-ones/

class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
    	int cur = 0, res = 0;
    	for (auto x : nums) {
    		if (x == 1) {
    			res = max(res, ++cur);
    		} else {
    			cur = 0;
    		}
    	}
    	return res;
    }
};